### PR TITLE
Support resolve override from open flag

### LINKED ISSUE #14178: Alluxio Fuse对接NFS
**Alluxio Version:** 2.6.0
What version of Alluxio are you using?

**Describe the bug**
启动Alluxio，然后利用fuse挂载本地的一个目录。再通过NFS把这个目录共享给另外一台Linux。在这两个目录下，可以正常查看Alluxio的数据。对文件进行ls、cat、rm等操作能够同步响应到Alluxio中。但是在NFS共享的目录用 echo“test”> test.txt写入文件，会发现目录可以创建这个文件，Alluxio fuse挂载的目录和Alluxio也同样可以看到该文件，但是文件内容为空。在Alluxio fuse挂载的目录上执行echo “test” >test 就可以正常写入。
通过Debug代码，发现两个目录的写入的流程并不一样。Alluxio fuse挂载的目录正常写入流程是这样的：
       Fuse.Getattr
       Fuse.Create
       Fuse.Getattr
       Fuse.Flush
       Fuse.Write
       Fuse.Flush
       Fuse.Getattr
       Fuse.Release
而挂载的共享目录的写入流程则是：
       Fuse.Getattr
       Fuse.Getattr
       Fuse.Create
       Fuse.Getattr
       Fuse.Release
       Fuse.Getattr
       Fuse.Open
       Fuse.Getattr
       Fuse.Write
       Fuse.Getattr
       Fuse.Release
       Fuse.Getattr
好像是在写入的时候先Create然后Release，导致了Write的过程中mCreateFileEntries无法找到对应的文件，就会报出can not find fd in the table. 我尝试修改了Open方法中的OverWrite的值，将其设置为true，让Alluxio删除这个文件然后重新Create一下，发现文件的内容是可以正常写入了。

**To Reproduce**
假设Alluxio集群中的一台服务器作为Server A，一台额外的服务器作为Server B。操作系统为Centos 7.5，Alluxio版本为2.6.0。

Server A 创建一个本地目录/mnt/nfs_a，Server B创建一个目录/mnt/nf_b。

#### Step 1: Alluxio-Fuse挂载

利用alluxio-fuse脚本挂载本地目录：

```shell
$/alluxio/integration/fuse/bin/alluxio-fuse/alluxio-fuse mount /mnt/nfs_a
Starting AlluxioFuse process: mounting alluxio path "/" to local mount point "/mnt/nfs_a" with options="big_writes"
Successfully mounted Alluxio path "/" to /mnt/nfs_a.
See /home/alluxio/IdeaProjects/alluxio/logs/fuse.log for logging messages
```

#### Step 2: NFS挂载

Server A 和Server B安装NFS工具，并启动这NFS服务。

```shell
$ yum install nfs-utils
$ systemctl start nfs
$ systemctl start rpcbinid
```

将Server A的NFS作为服务端，并将该机器下的/mnt/nfs_a作为共享目录共享出去。
编辑Server A中的/etc/exports

```shell
$ vim /etc/exports
#输入以下内容
/mnt/nfs_a *(rw,no_root_squash,fsid=0,sync)
#保存退出
```

在Server B中确认Server A的目录是否共享

```shell
$ showmount -e Server A ip
Export list for Server A ip:
/mnt/nfs_a *
```

在Server B中将Server A中的目录挂载到/mnt/nfs_b

```shell
$ mount -t nfs4 10.121.24.216:/ /mnt/nfs_b
```

此时，两个服务器中对应的目录均能够正常显示出Alluxio的数据。

**Expected behavior**
目前采用NFS对接Alluxio，期望能够在NFS的共享目录中对文件正常的读写。之后还需要利用Samba对接Alluxio，同样希望能够正常的读写文件。

### PR DESCRIPTION
**Pull request by maobaolong**
Fix #14178

### CODE REVIEW

**Comment by maobaolong** [comment]
@LuQQiu Would you please take a look at this PR? Thanks.

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java [hunk 1]
```diff
@@ -336,7 +336,10 @@ private int readdirInternal(String path, long buff, long filter, long offset,
   @Override
   public int open(String path, FuseFileInfo fi) {
     final int flags = fi.flags.get();
-    boolean overwrite = OpenFlags.valueOf(flags) == OpenFlags.O_WRONLY;
+    // TODO(maobaolong): Add an option to decide whether reject rw flag
+    //  or fallback to truncate
+    boolean overwrite = OpenFlags.valueOf(flags) == OpenFlags.O_WRONLY
+        || (flags & 0b11) != 0;
```
Comment (LuQQiu) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:342]: can you add a description for what 0b11 means?

Comment (LuQQiu) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:342]: (flags & 0b11) != 0; why doing this operation?
what flags included etc

Comment (yyongycy) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:342]: Wondering if the flags falls into one enum, comparing the enum, that could be more clear.

**Review by yyongycy — APPROVED** [review]
LGTM after adding some comments.

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**